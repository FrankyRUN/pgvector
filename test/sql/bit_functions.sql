SELECT hamming_distance(B'111', B'111');
SELECT hamming_distance(B'111', B'110');
SELECT hamming_distance(B'111', B'100');
SELECT hamming_distance(B'111', B'000');
SELECT hamming_distance(B'111', B'00');
SELECT hamming_distance(B'111', B'000'::varbit(4));

SELECT jaccard_distance(B'1111', B'1111');
SELECT jaccard_distance(B'1111', B'1110');
SELECT jaccard_distance(B'1111', B'1100');
SELECT jaccard_distance(B'1111', B'1000');
SELECT jaccard_distance(B'1111', B'0000');
SELECT jaccard_distance(B'1100', B'1000');
SELECT jaccard_distance(B'1111', B'000');
SELECT jaccard_distance(B'1111', B'0000'::varbit(5));
