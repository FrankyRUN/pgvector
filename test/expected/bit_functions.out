SELECT hamming_distance(B'111', B'111');
 hamming_distance 
------------------
                0
(1 row)

SELECT hamming_distance(B'111', B'110');
 hamming_distance 
------------------
                1
(1 row)

SELECT hamming_distance(B'111', B'100');
 hamming_distance 
------------------
                2
(1 row)

SELECT hamming_distance(B'111', B'000');
 hamming_distance 
------------------
                3
(1 row)

SELECT hamming_distance(B'111', B'00');
ERROR:  different bit lengths 3 and 2
SELECT hamming_distance(B'111', B'000'::varbit(4));
 hamming_distance 
------------------
                3
(1 row)

SELECT hamming_distance(B'111', B'0000'::varbit(4));
ERROR:  different bit lengths 3 and 4
SELECT jaccard_distance(B'1111', B'1111');
 jaccard_distance 
------------------
                0
(1 row)

SELECT jaccard_distance(B'1111', B'1110');
 jaccard_distance 
------------------
             0.25
(1 row)

SELECT jaccard_distance(B'1111', B'1100');
 jaccard_distance 
------------------
              0.5
(1 row)

SELECT jaccard_distance(B'1111', B'1000');
 jaccard_distance 
------------------
             0.75
(1 row)

SELECT jaccard_distance(B'1111', B'0000');
 jaccard_distance 
------------------
                1
(1 row)

SELECT jaccard_distance(B'1100', B'1000');
 jaccard_distance 
------------------
              0.5
(1 row)

SELECT jaccard_distance(B'1111', B'000');
ERROR:  different bit lengths 4 and 3
SELECT jaccard_distance(B'1111', B'0000'::varbit(5));
 jaccard_distance 
------------------
                1
(1 row)

SELECT jaccard_distance(B'1111', B'00000'::varbit(5));
ERROR:  different bit lengths 4 and 5
